### PR TITLE
leave a jupyter page if cluster stopped

### LINKED ISSUE #1148: User pauses or deletes cluster while in notebook or terminal, should be returned to workspace
Currently when a user pauses or deletes their cluster while in a notebook, they remain in the notebook and gradually see auto-saving errors:

Same thing happens if a user is in the terminal and they pause or delete a cluster:

Each of these cases should be handled for a better UX.

### PR DESCRIPTION
**Pull request by zarsky-broad**
(or deleted)

closes #1148

### CODE REVIEW

File: src/components/ClusterManager.js [hunk 1]
```diff
@@ -262,6 +262,15 @@ export default ajaxCaller(class ClusterManager extends PureComponent {
     }
   }
 
+  async executeAndRefreshWithNav(promise) {
+    const { namespace, name } = this.props
+
+    this.executeAndRefresh(promise)
+    if (/notebooks\/.+/.test(window.location.hash)) {
```
Comment (katevoss) [inline src/components/ClusterManager.js:269]: What does this line do?

Comment (zarsky-broad) [reply src/components/ClusterManager.js:269]: It makes the navigation happen only if you're on a notebook or terminal, not the notebooks list or anywhere else

**Comment by breilly2** [comment]
Sorry I didn't catch this before and comment on the issue. Is this the behavior we want? Could we instead give the user the option of restarting the cluster or capturing something from their work before we force them out? I worry about people losing something that they're in the middle of.

**Comment by zarsky-broad** [comment]
@breilly this is only if they click they stop or delete button.

**Comment by breilly2** [comment]
Ah, I get it now. Thanks!

**Review by breilly2 — APPROVED** [review]
👍

**Review by katevoss — APPROVED** [review]
Tested and works as expected

**Pull request merged**